manim/mobject/logo.py

```diff
@@ -0,0 +1,116 @@
+from ..animation.fading import FadeIn
+from ..animation.transform import ApplyMethod
+from ..mobject.geometry import Circle, Square, Triangle
+from ..mobject.svg.tex_mobject import Tex, MathTex
+from ..mobject.types.vectorized_mobject import VGroup
+from ..utils.tex_templates import TexFontTemplates
+
+
+class ManimBanner(VGroup):
+    r"""Convenience class representing Manim's banner.
+
+    Can be animated using custom methods.
+
+    Examples
+    --------
+
+    .. manim:: BannerDarkBackground
+
+        class BannerDarkBackground(Scene):
+            def construct(self):
+                banner = ManimBanner().scale(0.5).to_corner(DR)
+                self.play(FadeIn(banner))
+                self.play(banner.expand())
+                self.play(FadeOut(banner))
+
+    .. manim:: BannerLightBackground
+
+        class BannerLightBackground(Scene):
+            def construct(self):
+                self.camera.background_color = "#ece6e2"
+                banner = ManimBanner(dark_theme=False).scale(0.5).to_corner(UR)
+                self.play(FadeIn(banner))
+                self.play(banner.expand())
+                self.play(FadeOut(banner))
+
+    """
+
+    def __init__(self, dark_theme=True):
+        VGroup.__init__(self)
+
+        logo_green = "#81b29a"
+        logo_blue = "#454866"
+        logo_red = "#e07a5f"
+        m_height_over_anim_height = 0.75748
+
+        self.font_color = "#ece6e2" if dark_theme else "#343434"
+        self.scale_factor = 1
+
+        self.M = MathTex(r"\mathbb{M}").scale(7).set_color(self.font_color)
+        self.M.shift(2.25 * LEFT + 1.5 * UP)
+
+        self.circle = Circle(color=logo_green, fill_opacity=1).shift(LEFT)
+        self.square = Square(color=logo_blue, fill_opacity=1).shift(UP)
+        self.triangle = Triangle(color=logo_red, fill_opacity=1).shift(RIGHT)
+        self.add(self.triangle, self.square, self.circle, self.M)
+        self.move_to(ORIGIN)
+
+        anim = VGroup()
+        for i, ch in enumerate("anim"):
+            tex = Tex(
+                "\\textbf{" + ch + "}",
+                tex_template=TexFontTemplates.gnu_freeserif_freesans,
+            )
+            if i != 0:
+                tex.next_to(anim, buff=0.01)
+            tex.align_to(self.M, DOWN)
+            anim.add(tex)
+        anim.set_color(self.font_color).set_height(
+            m_height_over_anim_height * self.M.get_height()
+        )
+
+        self.anim = anim
+        self.anim.set_opacity(0)
+
+    def scale(self, scale_factor, **kwargs):
+        self.scale_factor *= scale_factor
+        self.anim.scale(scale_factor, **kwargs)
+        return super().scale(scale_factor, **kwargs)
+
+    def expand(self):
+        """An animation transforming the banner from its initial state (just icons)
+        to its expanded state (showing the full name and the icons).
+
+        See the class documentation for how to use this.
+        """
```

Comment:
Suggestion: please add a "Returns" section.
Suggestion (non-blocking): if possible, I'd love to have a single-line short summary (currently it is two lines...)

Reply:
Suggestion: please add a "Notes" section that mentions the fact that after the call to expand, "anim" will be part of the VGroup (it usually is not part of the VGroup).

Reply:
Improved overall documentation via 3fac1966